src/http-gateways/trustless-gateway.md

```diff
@@ -302,31 +302,73 @@ of their presence in the DAG or the value assigned to the "dups" parameter, as
 
-The metadata block is a regular CAR block with the following properties:
+The only supported value for the data type parameter is `json`. This signifies that the metadata MUST be a JSON object.
 
-- CID specifies multicodec `car-metadata` (`0x04ff`), see
-  [multicodec#334](https://github.com/multiformats/multicodec/pull/334).
+This parameter MUST only be used with CAR `version=1`.
 
-- The payload contains metadata encoded as DAG-CBOR.
+When the parameter is not set or does not equal `eof+json`, the server SHOULD not add any extra blocks to the response, neither the 0x00 byte nor any metadata.
 
-The metadata MUST include the following fields:
+When `meta=eof+json`, the JSON object SHOULD conform to the following [JSON schema](https://json-schema.org/).
 
-- `len` - byte length of the CAR data (excluding the metadata block)
-- `b3h` - Blake3 hash (checksum) of the CAR data (excluding the metadata block).
-- `b3h_sig` - A signature over `<len><b3h><request>` using server's Ed2559 identity.
-  - `len` is encoded as `varint`,
-  - `b3h` is encoded as 32 bytes,
-  - The effective query as executed by the gateway. This query is the request url - path and query string arguments.
+```json
+{
+  "properties": {
+    "description": "Properties of the response"
+    "type": "object"
+  },
+  "error": {
+    "description": "Error message"
+    "type": "string"
+  },
+  "sig": {
+    "description": "A signature, using the server's Ed2559 identity, over the metadata properties object"
+    "type": "string"
+  }
+}
+```
+
+The properties object can include any fields that the server would like to implement. The following properties fields are mentioned explicitly to reach a convention on their definition as they have existing use cases.
+
+```json
+{
+  "car_bytes": {
+    "description": "The total byte length of the CAR stream (excluding the 0x00 byte and the metadata block)",
+    "type": "integer"
+  },
+  "data_bytes": {
+    "description": "Total byte length of blocks (excluding the 0x00 byte and the metadata block, but including duplicates when present)",
+    "type": "integer"
+  },
+  "block_count": {
+    "description": "Total number of blocks present in the CAR stream (excluding the 0x00 byte and the metadata block, but including duplicates when present)",
+    "type": "integer"
+  },
+  "car_cid": {
+    "description": "A hash of the CAR stream giving a CIDv1 with 0x0202 codec",
+    "type": "string"
+  },
+  "b3checksum": {
+    "description": "A Blake3 hash (checksum) of the CAR stream (excluding the 0x00 byte and the metadata block)",
+    "type": "string"
+  },
+  "content_path": {
+    "description": "The url path in the request as executed by the gateway",
+    "type": "string"
+  },
+  "query_params": {
+    "description": "The query string in the request as executed by the gateway",
+    "type": "string"
+  }
```

Comment:
Based on the discussion in https://github.com/ipfs/specs/pull/431#discussion_r1327717940, I think we should choose different names here.

How about simplifying things and merging these two fields into one?

```suggestion
  "retrieval_params": {
    "description": "Retrieval parameters describing what the client requested from the gateway",
    "type": "object"
  }
```